x-pack/packages/ml/data_grid/components/data_grid.tsx

```diff
@@ -35,18 +36,49 @@ import { isPopulatedObject } from '@kbn/ml-is-populated-object';
 import { euiDataGridStyle, euiDataGridToolbarSettings, INDEX_STATUS } from '../lib/common';
 import { UseIndexDataReturnType } from '../lib/types';
 
-const cssOverride = css({
-  '.euiDataGridRowCell--boolean': { textTransform: 'none' },
-  // Overrides to align the sorting arrow, actions icon and the column header when no chart is available,
-  // to the bottom of the cell when histogram charts are enabled.
-  // Note that overrides have to be used as currently it is not possible to add a custom class name
-  // for the EuiDataGridHeaderCell - see https://github.com/elastic/eui/issues/5106
-  '.euiDataGridHeaderCell': {
-    '.euiDataGridHeaderCell__sortingArrow,.euiDataGridHeaderCell__icon,.euiPopover': {
-      marginTop: 'auto',
+const histogramHeaderProps = { className: 'cellHeaderWithHistogramChart' };
+const cssOverride = ({ euiTheme }: UseEuiTheme) =>
+  ({
+    '.euiDataGridRowCell--boolean': { textTransform: 'none' },
+    // When a histogram chart is present in the column header, absolutely position
+    // the sorting arrow and actions icon so that the chart can expand to the full width
+    // of the header
+    '.cellHeaderWithHistogramChart': {
```

Comment:
@elastic/ml-ui - I had to update a good amount of custom CSS to account for the histogram chart in the column header and the header redesign (see https://github.com/elastic/eui/pull/7371 for more info, or the PR description for more screenshots). Here's what the final behavior looks like:

![ml](https://github.com/elastic/kibana/assets/549407/007032ce-77ac-47c9-858c-d66f192e66d1)

I'm not as familiar with your UI and its edge cases however, and would appreciate your team QAing it to make sure everything looks/works as expected.

cc @MichaelMarcialis and @andreadelrio just as an FYI on this, to let y'all know how the new design affects some custom implementations (or vice versa)